Jingle/SCJingleConverter/composition.cpp

```diff
@@ -849,37 +859,137 @@ uint32_t Composition::getMemUsage() {
+                if (chordIdxL < note.frequencies.size()) {
+                    note_frequency = note.frequencies[chordIdxL];
+                }
+                if (chordIdxL < tie.frequencies.size()) {
+                    tie_frequency = tie.frequencies[chordIdxL];
+                }
+
+                if (note_frequency == tie_frequency) {
+                    note.length += tie.length;
+                    optimizedNotesL.erase(optimizedNotesL.begin()+notes_idx+1);
+                    removed_notes++;
+                }
+            }
         }
-        break;
-    }
 
-    if (voice) {
-        if (meas_start_idx >= voice->measures.size() || meas_end_idx >= voice->measures.size()) {
-            qDebug() << "Start or End Measure Index out of bounds for Channel in getNumNotes()";
-            return 0;
+        // Remove trailing the trailing rest (should only be one max)
+        if (optimizedNotesL[optimizedNotesL.size()-1].frequencies[chordIdxL] == 0.f) {
+            optimizedNotesL.erase(optimizedNotesL.begin()+optimizedNotesL.size()-1);
         }
+    }
 
+    // Optimize Notes in Right Channel
+    if (voices.count(voiceStrR)) {
+        // Collect notes into single vector
         for (uint32_t meas_idx = meas_start_idx; meas_idx <= meas_end_idx; meas_idx++) {
-            Measure& meas = voice->measures[meas_idx];
-            num_notes += meas.notes.size();
+            Measure& meas = voices[voiceStrR].measures[meas_idx];
+            for (uint32_t notes_idx = 0; notes_idx < meas.notes.size(); notes_idx++) {
+                optimizedNotesR.push_back(meas.notes[notes_idx]);
+            }
+        }
+
+        // Combine ties
+        for (uint32_t notes_idx = optimizedNotesR.size()-1; notes_idx-- > 0; ) {
+            Note& note = optimizedNotesR[notes_idx];
+
+            if (note.tied && notes_idx < optimizedNotesR.size()-1) {
+                Note& tie = optimizedNotesR[notes_idx+1];
+
+                float note_frequency = note.frequencies.back();
+                float tie_frequency = tie.frequencies.back();
+                if (chordIdxR < note.frequencies.size()) {
+                    note_frequency = note.frequencies[chordIdxR];
+                }
+                if (chordIdxR < tie.frequencies.size()) {
+                    tie_frequency = tie.frequencies[chordIdxR];
+                }
+
+                if (note_frequency == tie_frequency) {
+                    note.length += tie.length;
+                    optimizedNotesR.erase(optimizedNotesR.begin()+notes_idx+1);
+                    removed_notes++;
+                }
+            }
+        }
+
+        // Remove trailing the trailing rest (should only be one max)
+        if (optimizedNotesR[optimizedNotesR.size()-1].frequencies[chordIdxR] == 0.f) {
+            optimizedNotesR.erase(optimizedNotesR.begin()+optimizedNotesR.size()-1);
+            removed_notes++;
+        }
+    }
+
+    // Remove leading silence
```

Comment:
Add a GUI control to make this optional. I'm all for changing the internal structure to save space, but don't want to force any timing changes upon a user (maybe some people want preceding silence?).